src/Style.hs

```diff
@@ -283,12 +269,40 @@ attribute = identifier -- TODO: Naming convention - same as identifiers?
 -- styPrettyPrint b = ""
 
 --------------------------------------------------------------------------------
--- Functions used by "Runtime"
+-- Functions used by "Runtime" -- Style Translator
+-- Translates from a Style AST to:
+-- * a list of geometric primitives
+-- * a list of constraints
+-- * a list of objectives
+
+-- Type aliases for readability in this section
+-- | 'StyContext' maintains the current output of the translater
+type StyContext a =
+    (StyDict, -- | dictionary mapping Substance ID to Style objects
+    [(ObjFnOn a, Weight a, [Name], [a])],    -- | List of objective functions
```

Comment:
Why not use `ObjFnInfo a` and `ConstrFnInfo a`?

Reply:
Oh true. Pushing a quick fix now. Any other suggestion?